valohai_yaml/utils/merge.py

```diff
@@ -0,0 +1,37 @@
+import copy
+
+MISSING = object()
+
+
+def _merge_dicts(
+        a: dict,
+        b: dict,
+        merger,
+        copier=lambda v: v,
+        skip_missing_a: bool = False,
+        skip_missing_b: bool = False) -> dict:
+    out = {}
```

Comment:
This could use the somewhat esoteric incantation `out = type(a)()` to create an object of the same type as the original. That'd allow one to pass in an `OrderedDict` and get one out too.